AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -484,7 +482,7 @@ am_Error_e CAmSocketHandler::removeSignalHandler(const sh_pollHandle_t handle)
     {
         if (it->handle == handle)
         {
-            it = mSignalHandlers.erase(it);
+            mSignalHandlers.erase(it);
```

Comment:
This still doesn't feel quite right to me...  I can see that it probably works, when mSignalHandlers is a vector.  But (and I did some tests) the previous code seems to also work (because the "next" iterator that is returned is in fact the same pointer as it was because the data is moved down one step during erase).  Now I would agree is a bit unexpected that the previous code on the left works correctly (on vectors).   

But in the new code, if the container type is ever changed to a linked list the behavior is undefined (will crash) since the iterator is invalidated by erase.

It would be better to apply a safe pattern for all similar cases.  Maybe rewrite with the **remove**() algorithm and a functor instead?
Or: https://stackoverflow.com/questions/2874441/deleting-elements-from-stl-set-while-iterating

Reply:
Since we are directly return in case of erase the iterator element will be not used anymore.
The assignment is not needed.

--> Refuse change request

Reply:
Well that's also what I concluded.  It should work, but it's not following the normal pattern of erase, which is to remember the iterator.  I guess what I am after is that for this for all other similar cases, use the same type of loop (probably a while loop), even if there are some cases where there's a return.

(You can refuse any of my requests for change, of course)

Reply:
BTW: I did it to keep a common code style.